src/h3/_cy/cells.pyx

```diff
@@ -125,25 +129,22 @@ cpdef H3int[:] ring(H3int h, int k):
     """ Return cells at grid distance `== k` from `h`.
     Collection is "hollow" for k >= 1.
     """
-    cdef:
-        h3lib.H3Error err
-
     check_cell(h)
     check_distance(k)
 
     n = 6*k if k > 0 else 1
     ptr = create_ptr(n)
 
-    err = h3lib.gridRingUnsafe(h, k, ptr)
-
-    # if we drop into the failure state, we might be tempted to not create
-    # this mv, but creating the mv is exactly what guarantees that we'll free
-    # the memory. context manager would be better here, if we can figure out
-    # how to do that
-    mv = create_mv(ptr, n)
-
-    if err:
+    try:
+        check_for_error(
+            h3lib.gridRingUnsafe(h, k, ptr)
+        )
+    except H3ValueError:
```

Comment:
I would expect try/except to be relatively slow here compared to looking at the error value itself. That may not matter for this PR.

Reply:
I thought about it, and I think I've got a better approach with these latest changes. Let me know what you think!